packages/terra-table/src/Table.jsx

```diff
@@ -1,68 +1,401 @@
+import Cell from './subcomponents/_Cell';
+import Section from './subcomponents/_Section';
+import HeaderRow from './subcomponents/_HeaderRow';
+import HeaderCell from './subcomponents/_HeaderCell';
+import ChevronCell from './subcomponents/_ChevronCell';
+import CheckMarkCell from './subcomponents/_CheckMarkCell';
+import HeaderChevronCell from './subcomponents/_HeaderChevronCell';
+import HeaderCheckMarkCell from './subcomponents/_HeaderCheckMarkCell';
 
 const cx = classNames.bind(styles);
 
 const propTypes = {
   /**
-   * The children passed to the component
+   * The check mark styling to apply.
+   */
+  checkStyle: PropTypes.oneOf([
+    'none',
+    'readOnly', // TODO: address renaming this, UX ask
+    'toggle',
+  ]),
+  /**
+   * The divider styling to apply to the child rows.
+   */
+  dividerStyle: PropTypes.oneOf(['none', 'vertical', 'horizontal', 'both']),
+  /**
+   * The width value structures associated to each column.
+   */
+  columnWidths: PropTypes.arrayOf(widthShape),
+  /**
+   * Whether or not the rows should have chevrons applied.
+   */
+  hasChevrons: PropTypes.bool,
+  /**
+   * The data to build header cells and columns.
+   */
+  headerData: headerShape,
+  /**
+   * Element to append to the top of the table. i.e. toolbars etc.
+   */
+  headerNode: PropTypes.node,
+  /**
+   * Whether or not the table should expanded to fill its parent element.
+   */
+  fill: PropTypes.bool,
+  /**
+   * Element to append to the bottom of the table. i.e. toolbars etc.
    */
-  children: PropTypes.node.isRequired,
+  footerNode: PropTypes.node,
   /**
-   * Whether or not the rows should be zebra striped
+   * This value is used for accessibility when paged/virtualized rows are used.
+   * By default this value is derived from the number of rows passed within the sectionData.
    */
-  isStriped: PropTypes.bool,
+  numberOfRows: PropTypes.number,
   /**
-   * Whether or not the table cells should be padded
+   * The padding styling to apply to the cell content.
    */
-  isPadded: PropTypes.bool,
+  paddingStyle: PropTypes.oneOf(['none', 'standard', 'compact']),
+  /**
+   * The interaction styling to apply to the row.
+   */
+  rowStyle: PropTypes.oneOf([
+    'none',
+    'disclose',
+    'toggle',
+  ]),
+  /**
+   * Function callback returning the html node of the table's inner body element.
+   */
+  scrollRefCallback: PropTypes.func,
+  /**
+   * The grouping of rows associated to the table. A section header can also be paired with the rows.
+   */
+  sectionData: PropTypes.arrayOf(sectionShape),
+  /**
+   * The summary text to describe the table's content and interactions.
+   */
+  summary: PropTypes.string.isRequired,
+  /**
+   * The element id to associate to the descriptive text.
+   */
+  summaryId: PropTypes.string.isRequired,
 };
 
 const defaultProps = {
-  isStriped: true,
-  isPadded: true,
+  dividerStyle: 'none',
+  fill: false,
+  paddingStyle: 'none',
+  rowStyle: 'none',
+  checkStyle: 'none',
```

Comment:
Looking at these `none` values here makes we wonder if we could just use an undefined value as `none` and drop explicit `none` values from the API?